HashTable/main.c

```diff
@@ -52,7 +52,7 @@ int main(int argc, char ** argv){
 	printf("ht[\"7\"]: %s\n", (const char *)hashtable_get(ht, (void *)"7"));
 	printf("ht[\"8\"]: %s\n", (const char *)hashtable_get(ht, (void *)"8"));
 
-	hashtable_destroy(ht);
+	hashtable_destroy(ht);  // why this changes are made to this i want to know .. help me out
```

Comment:
I'm not sure what you are asking but, the 'hashtable_destroy' function call will loop through the entire data structure and deallocate memory previously allocated by each insertion (or resize), then it will free any remaining memory the hashtable itself is using.

This isn't necessary when just exiting afterwards since all the memory will be free'd anyways.  But freeing up memory is always a good idea to do when at all possible otherwise the application will become bloated and could incur unexpected problems.